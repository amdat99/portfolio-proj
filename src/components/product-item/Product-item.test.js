import React from 'react';

import { shallow,render,mount } from 'enzyme';
import {createSerializer} from 'enzyme-to-json';
import toJson from 'enzyme-to-json';


import ProductItem from './Product-item'

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));
// describe('cart item', () => {
      
        let mockFetchProductPending = jest.fn();
        let  mockIncrementItem = jest.fn();
        let mockDecrementItem = jest.fn();
        let wrapper
        let wrapper2

        
  beforeEach(() => {

   const mockProps = {
            item: {
              picture: 'www.picture.com',
              price: 10,
              name: 'ball',
              quantity: 5,
              profileId: 123
            },
           
            incrementItem: mockIncrementItem,
          
          }; 
          
           wrapper = toJson(shallow(<ProductItem  {...mockProps} />))
           wrapper2 = shallow(<ProductItem {...mockProps} />);
        })
        



it('expect to render cartItem component', () => {
    expect(wrapper).toMatchSnapshot();
})


  it('expect increment item to be called when item added', () => {
    wrapper2.find('.product-item-button').simulate('click')
    expect(mockIncrementItem).toHaveBeenCalled();
  });
;
