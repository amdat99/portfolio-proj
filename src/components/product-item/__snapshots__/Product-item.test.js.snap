// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render cartItem component 1`] = `
<div
  id="product-item-container"
>
  <h3
    id="product-item-name"
  >
    BALL
  </h3>
  <img
    alt="item"
    id="product-item-image"
    src="www.picture.com"
  />
  <div
    id="product-item-overlay"
  >
    <div
      id="product-item-labels"
    >
      <span>
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
      </span>
      <h2
        id="product-item-price"
      >
        Â£
        10
      </h2>
      <span
        id="product-item-seller"
      >
        seller: 
      </span>
      <button
        className="product-item-button"
        onClick={[Function]}
      >
        Add to cart
      </button>
    </div>
  </div>
</div>
`;
